geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync, unlinkSync } from 'fs';
+import { Extract } from 'unzipper';
+const mapshaper = require('mapshaper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          Extract({
+            path: `${fileInfo.destination}/${fileInfo.filename.replace(
+              '.zip',
+              '',
+            )}`,
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
+        )
+        .on('error', (error: Error) =>
+          reject(
+            new Error(
+              `${fileInfo.filename} could not been extracted: ` + error,
+            ),
+          ),
+        );
+    });
+  }
+
+  private async shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    const outputKey = `shapefile-${new Date().getTime()}.geojson`;
+
+    const _geoJson = await mapshaper.applyCommands(
+      `-i ${fileInfo.path.replace(
+        '.zip',
+        '',
+      )}/*.shp -info -o ${outputKey} -verbose `,
+    );
+
+    return JSON.parse(_geoJson[outputKey].toString('utf-8'));
+  }
+
+  private deleteShapefileData(path: string): void {
+    if (path.startsWith('/tmp')) {
+      unlinkSync(path);
+      rmdirSync(path.replace('.zip', ''), { recursive: true });
```

Comment:
if you don't want to block the event loop, you may consider using `unlink` and `rmdir` from `fs/promises`